### PR TITLE
✨ Mongo compendium data, automatic reloading

### PR DESCRIPTION
**Pull request by madcowfred**
### Summary
Use MongoDB as the source of compendium data

#### PR Type

- [ ] This PR is a code change that implements a feature request.
- [ ] This PR fixes an issue.
- [x] This PR adds a new feature that is not an open feature request.
- [ ] This PR is not a code change (e.g. documentation, README, ...)
#### Other
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] If code changes were made then they have been tested.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

File: cogs5e/funcs/lookupFuncs.py [hunk 1]
```diff
@@ -25,31 +26,98 @@
...
+        self.srd_races = self.read_json('srd-races.json', [])
+        self.srd_spells = self.read_json('srd-spells.json', [])
+
+        # Dictionary!
+        self.itemprops = self.read_json('itemprops.json', {})
+
+        self.load_common()
+
+    async def load_all_mongodb(self, mdb):
+        lookup = {d.key: d.object for d in await mdb.static_data.find({}).to_list(length=None)}
```
Comment (mommothazaz123) [inline cogs5e/funcs/lookupFuncs.py]: This should probably be `d['key']: d['object']`

Comment (madcowfred) [reply cogs5e/funcs/lookupFuncs.py]: Yup, found that out when I actually imported some data. Oops.

File: cogs5e/funcs/lookupFuncs.py [hunk 2]
```diff
@@ -25,31 +26,98 @@
...
+
+    async def reload_task(self, mdb=None):
+        wait_for = int(os.getenv('RELOAD_INTERVAL', '300'))  # TODO: decide if 5 minutes is a reasonable default
+        if wait_for > 0:
+            log.info("Reloading data every {} seconds", wait_for)
+            while True:
+                await self.reload(mdb)
+                await asyncio.sleep(wait_for)
+
+    async def reload(self, mdb=None):
```
Comment (mommothazaz123) [inline cogs5e/funcs/lookupFuncs.py]: Just declaring the reload function async doesn't actually make it run in an executor by default - to do that, you'd want to make all the blocking calls a sync function (specifically L64-77 and `load_common()`), then call `loop.run_in_executor()`.

Check out utils/functions.py L500 (`generate_token()`) for an example

Comment (madcowfred) [reply cogs5e/funcs/lookupFuncs.py]: I miss so many Github notifications, oops.

The reload function should only ever be run in an executor, I... think? It's called from reload_task() which is started in bot.py as `bot.loop.create_task(compendium.reload_task(bot.mdb))`, and also called from a bot command. If there's extra work to do then Python async is even weirder than I thought!

Comment (madcowfred) [reply cogs5e/funcs/lookupFuncs.py]: Oh dear, it is weirder than I thought.

File: cogsmisc/adminUtils.py [hunk 3]
```diff
@@ -155,6 +156,12 @@ def __init__(self, bot):
         await self.bot.change_presence(status=status, activity=discord.Game(msg or "D&D 5e | !help"))
         await ctx.send("Changed presence.")
 
+    @commands.command(hidden=True)
+    @checks.is_owner()
+    async def reload_static(self, ctx):
+        await compendium.reload(ctx.mdb)
```
Comment (mommothazaz123) [inline cogsmisc/adminUtils.py]: should probably be `self.bot.mdb`

**Pull request merged**